src/hooks/worlds/useOwnWorlds.ts

```diff
@@ -0,0 +1,31 @@
+import { useFirestore, useFirestoreCollectionData } from "reactfire";
+import { collection, query, where } from "firebase/firestore";
+
+import { ALWAYS_EMPTY_ARRAY, COLLECTION_WORLDS } from "settings";
+
+import { World } from "api/world";
+
+import { CONVERTER_WORLD_WITH_ID } from "utils/converters";
+import { WithId } from "utils/id";
+
+import { useUser } from "hooks/useUser";
+
+export const useOwnWorlds = (): WithId<World>[] => {
+  const firestore = useFirestore();
+  const { userId } = useUser();
+
+  const ownWorldsRef = query(
+    collection(firestore, COLLECTION_WORLDS),
+    where("isHidden", "==", false),
+    where("owners", "array-contains", userId ?? "")
+  ).withConverter(CONVERTER_WORLD_WITH_ID);
+
+  const { data: ownWorlds } = useFirestoreCollectionData<WithId<World>>(
+    ownWorldsRef,
+    {
+      initialData: ALWAYS_EMPTY_ARRAY,
+    }
+  );
```

Comment:
There is a `useFireCollection()` hook that might be used for DRY and more declarative code.
NOTE: `useRefiCollection()` uses `reactfire`, but `useFireCollection()` doesn't

Reply:
I don't think it brings much of value here based on its current implementation. I've noticed it had never been used before. I'd expect it to at least accept query, not path/constraints.

![image](https://user-images.githubusercontent.com/63194656/153308679-21674f59-991a-47f7-9957-30d6e2481508.png)